codelists/actions.py

```diff
@@ -110,17 +90,60 @@ def create_codelist_with_codes(
 
 @transaction.atomic
 def create_codelist_from_scratch(
-    *, owner, name, coding_system_id, draft_owner, slug=None
+    *,
+    owner,
+    draft_owner,  # The User who can edit the draft CodelistVersion
+    name,
+    coding_system_id,
+    slug=None,
+    tag=None,
+    description=None,
+    methodology=None,
+    references=None,
+    signoffs=None,
 ):
-    """Create a new Codelist with a draft CodelistVersion."""
+    """Create a new Codelist with a draft new-style version."""
+
+    codelist = _create_codelist_with_handle(
+        owner=owner,
+        name=name,
+        coding_system_id=coding_system_id,
+        description=description,
+        methodology=methodology,
+        slug=slug,
+        references=references,
+        signoffs=signoffs,
+    )
+    codelist.versions.create(draft_owner=draft_owner)
+    return codelist
+
 
+def _create_codelist_with_handle(
+    owner,  # Can be an Organisation or a User
+    name,
+    coding_system_id,
+    description,
+    methodology,
+    slug=None,
+    references=None,
+    signoffs=None,
+):
     if not slug:
         slug = slugify(name)
 
-    codelist = owner.codelists.create(
-        slug=slug, name=name, coding_system_id=coding_system_id
+    codelist = Codelist.objects.create(
+        coding_system_id=coding_system_id,
+        description=description,
+        methodology=methodology,
     )
-    codelist.versions.create(draft_owner=draft_owner)
+    owner.handles.create(codelist=codelist, slug=slug, name=name, is_current=True)
+
+    for reference in references or []:
```

Comment:
The `or []` here is a really nice touch for keeping the kwarg immutable 👍

Reply:
Setting the default to empty tuple (`references=()`) is another similar trick you can use in this context. Though here it would require using it as the default all the way up the call-chain so it would be more awkward.